components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -745,4 +745,17 @@ public static String resolveEventName(String notificationChannel) {
             return IdentityEventConstants.Event.TRIGGER_NOTIFICATION;
         }
     }
+
+    /**
+     * Validate email username.
+     *
+     * @param username Tenant aware username of the user.
+     * @throws IdentityRecoveryClientException If username is not an email when email username is enabled.
+     */
+    public static void validateEmailUsername(String username) throws IdentityRecoveryClientException {
+
+        if (IdentityUtil.isEmailUsernameEnabled() && StringUtils.countMatches(username, "@") != 1) {
```

Comment:
is it allowed to have @ character in email username? e.g s@thya@wso2.com? if so, the logic hear should be updated

Reply:
This depends on the user store regex configuration. Regex used for email username '@' is not allowed.

Reply:
if someone changes the regex to allow the @, this validation will fail right? so shall we change this as, `StringUtils.countMatches(username, "@") == 0`